islandora-preservica-validation/preservicaObjCapture.py

```diff
@@ -0,0 +1,179 @@
+def getRepresentionInfo(sUrl, sRef_ID):
+    stempUrl = sUrl + sRef_ID
+    testInfoObj = getObjInfo(stempUrl)
+    if len(testInfoObj) > 0:
+        try:
+            sRepUrl = next(iter(testInfoObj.values()))['representationInfo']
+            req_Rep = requests.get(sRepUrl, headers=headers)
+            req_Rep.raise_for_status()
+            xml_reqRep = str(req_Rep.content.decode('UTF-8')) 
+            representation_rep = etree.ElementTree(etree.fromstring(xml_reqRep))
+            representations = representation_rep.findall(
+                './/{http://preservica.com/EntityAPI/v7.2}Representations/{http://preservica.com/EntityAPI/v7.2}Representation'
+            )
+            contentObj_lst ={}
+            BitstreamCount =0
+            for representation in representations:
+                #print(representation.text, " ", representation.attrib)
+                
+                #get all content object IDs for each of the represention
+                listOfRepresentContent = getContentObjID(representation.text)
+                
+                if len(listOfRepresentContent) > 0 : #get value of the first key since only one element
+                    #iterate each contentObj to get its generations for computing bitstreams
+                    for co in listOfRepresentContent.values():
+                        if len(co) > 0:
+                            for i in co:
+                               BitstreamCount += getbitstreamInfo("generations", i)[i]
+                    #print("Total bitstream for :" ,next(iter(listOfRepresentContent)), "is ", BitstreamCount)
+
+                    if not (next(iter(listOfRepresentContent)) in contentObj_lst.keys()):   
+                        contentObj_lst[next(iter(listOfRepresentContent))] = BitstreamCount
+                    else:
+                        contentObj_lst[next(iter(listOfRepresentContent))] += BitstreamCount
+            return (contentObj_lst)    
+        except requests.exceptions.RequestException as e:
+            print("Error: ", e)  
+
+#match the count, if match update the xml, otherwise output log error
+def preservica_bitstream_valid (f_in):
+    #open file to read the output file from islandoraObjcheck eg. mscount.csv)
+    with open (os.path.join(f_path, f_in), 'r', newline='') as islandoraCount_f:
+        csvreader = csv.DictReader(islandoraCount_f)
+        
+        with open(os.path.join(f_path, valid_result_f), 'w', newline='') as result_f:
+            header_lst = ['PID', 'islandora_count', 'preservica_refID', 'bitstreamCount','isValid']
+            f_writer = csv.DictWriter(result_f, fieldnames=header_lst)
+            f_writer.writeheader()
+            
+            global st_timer, curr_session
+            for row in csvreader:
+                #logic to check token expiration
+                if (round((time.time()-st_timer)*10**3) - 600000 >0 ):  
+                   new_session = token_fn.getRefreshToken(curr_session)
```

Comment:
`getRefreshToken()` allows for an empty return in the case of a server error, but this condition is not caught here.